mosquitto/2.0.15.patch

```diff
@@ -0,0 +1,148 @@
+From a840ddcf4c7c75d5dc28a8c154b51fad9a39b29e Mon Sep 17 00:00:00 2001
+From: Eric Blankenhorn <eric@wolfssl.com>
+Date: Mon, 23 Jan 2023 08:20:51 -0600
+Subject: [PATCH] Support wolfssl instead of openssl
+
+Changes:
+- `config.mk`: add the `WITH_TLS=wolfssl` option
+- The`wolfssl/options.h` header is included by defining the `EXTERNAL_OPTS_OPENVPN` macro
+- `net_mosq.c`: UI_METHOD not implemented in wolfSSL
+- `net_mosq.h`: UI_METHOD not implemented in wolfSSL
+- `net_mosq_ocsp.c`: safestack.h not implemented in wolfSSL
+
+wolfSSL
+
+```
+git clone https://github.com/wolfSSL/wolfssl.git
+cd wolfssl
+./autogen.sh
+./configure --enable-opensslextra --enable-opensslall --enable-ocsp --enable-ocspstapling --enable-context-extra-user-data --enable-psk --enable-sessioncerts --enable-crl CFLAGS="-DOPENSSL_VERSION_NUMBER=0x10100000"
+make
+make install
+```
+
+Eclipse Mosquitto
+
+```
+git clone https://github.com/eclipse/mosquitto.git
+cd mosquitto
+patch -p1 < <path/to/patch/file>
```

Comment:
I am on v2.0.15 commit b0277869d9806f6fab8e1bc11c4a4987c9a79ded and I'm getting two rejected chunks:
```
patching file lib/net_mosq.c
Hunk #2 succeeded at 127 with fuzz 2.
Hunk #3 FAILED at 162.
Hunk #4 FAILED at 191.
2 out of 4 hunks FAILED -- saving rejects to file lib/net_mosq.c.rej

cat lib/net_mosq.c.rej
--- lib/net_mosq.c
+++ lib/net_mosq.c
@@ -162,8 +164,10 @@ void net__cleanup(void)
 #  endif
 
    CONF_modules_unload(1);
+#ifndef USE_WOLFSSL
    cleanup_ui_method();
 #endif
+#endif
 
 #ifdef WITH_SRV
    ares_library_cleanup();
@@ -191,7 +195,9 @@ void net__init_tls(void)
 #if !defined(OPENSSL_NO_ENGINE)
    ENGINE_load_builtin_engines();
 #endif
+#ifndef USE_WOLFSSL
    setup_ui_method();
+#endif
    if(tls_ex_index_mosq == -1){
        tls_ex_index_mosq = SSL_get_ex_new_index(0, "client context", NULL, NULL, NULL);
    }
```

Reply:
Wow... That took me way too long to figure out! The whitespace was getting changed in my patch file (tabs->spaces). I'll fix in the next push.